test/cli/registry_auth_test.go

```diff
@@ -0,0 +1,78 @@
+package cli
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestRegistryAuth(t *testing.T) {
+	tests := []struct {
+		name       string
+		args       []string
+		env        map[string]string
+		assertions []traitAssertion
+	}{
+		{
+			name: "fallback to keychain",
+			args: []string{"-vv", "registry:localhost:5000/something:latest"},
+			assertions: []traitAssertion{
+				assertInOutput("source=OciRegistry"),
+				assertInOutput("localhost:5000/something:latest"),
+				assertInOutput("no registry credentials configured, using the default keychain"),
+			},
+		},
+		{
+			name: "use creds",
```

Comment:
Two more cases here where password without username and username without password would be great

Reply:
I can add that on 👍